discoverd/client/deployment.go

```diff
@@ -0,0 +1,195 @@
+package discoverd
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
+	"github.com/flynn/flynn/pkg/attempt"
+	"github.com/flynn/flynn/pkg/stream"
+)
+
+type DeploymentState string
+
+const (
+	DeploymentStatePerforming DeploymentState = "performing"
+	DeploymentStateDone       DeploymentState = "done"
+)
+
+func NewDeployment(service string) (*Deployment, error) {
```

Comment:
This seems like it should be in a separate package, does it depend on any client internals?